From 6a37286c2add96dba42fdd285162228eb55a99f1 Mon Sep 17 00:00:00 2001
From: "Keith T. Garner" <kgarner@kgarner.com>
Date: Wed, 5 May 2021 09:04:15 -0500
Subject: [PATCH] kernel: add kmod-leds-uleds

The allows userspace LEDs to be created and controlled. This can be useful
for testing triggers and can also be used to implement virtual LEDs.

Signed-off-by: Keith T. Garner <kgarner@kgarner.com>
[squash fixup commit and improve option wording]
Signed-off-by: Paul Spooren <mail@aparcar.org>
---
 package/kernel/linux/modules/leds.mk | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/package/kernel/linux/modules/leds.mk b/package/kernel/linux/modules/leds.mk
index 4dde833664..fdd5280fe4 100644
--- a/package/kernel/linux/modules/leds.mk
+++ b/package/kernel/linux/modules/leds.mk
@@ -145,3 +145,17 @@ define KernelPackage/leds-pwm/description
 endef
 
 $(eval $(call KernelPackage,leds-pwm))
+
+define KernelPackage/leds-uleds
+  SUBMENU:=$(LEDS_MENU)
+  TITLE:=Userspace LEDs
+  KCONFIG:=CONFIG_LEDS_USER
+  FILES:=$(LINUX_DIR)/drivers/leds/uleds.ko
+  AUTOLOAD:=$(call AutoLoad,60,leds-uleds,1)
+endef
+
+define KernelPackage/leds-uleds/description
+ This option enables support for userspace LEDs.
+endef
+
+$(eval $(call KernelPackage,leds-uleds))
-- 
2.33.1

From 3e9318f3c0be1ae14d9d4eb705b02e7441d5d26e Mon Sep 17 00:00:00 2001
From: Evgeny Kolesnikov <evgenyz@gmail.com>
Date: Tue, 25 May 2021 16:38:33 +0200
Subject: [PATCH] kernel: fix AutoLoad parameter for uleds module

The name of the module is 'uleds', not 'leds-uleds'.

Signed-off-by: Evgeny Kolesnikov <evgenyz@gmail.com>
[improve commit title]
Signed-off-by: David Bauer <mail@david-bauer.net>
---
 package/kernel/linux/modules/leds.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package/kernel/linux/modules/leds.mk b/package/kernel/linux/modules/leds.mk
index 301d314b1d..0ef0aee3fb 100644
--- a/package/kernel/linux/modules/leds.mk
+++ b/package/kernel/linux/modules/leds.mk
@@ -151,7 +151,7 @@ define KernelPackage/leds-uleds
   TITLE:=Userspace LEDs
   KCONFIG:=CONFIG_LEDS_USER
   FILES:=$(LINUX_DIR)/drivers/leds/uleds.ko
-  AUTOLOAD:=$(call AutoLoad,60,leds-uleds,1)
+  AUTOLOAD:=$(call AutoLoad,60,uleds,1)
 endef
 
 define KernelPackage/leds-uleds/description
-- 
2.33.1

